bin/setenv.sh

```diff
@@ -0,0 +1,179 @@
+    CATALINA_OPTS="${CATALINA_OPTS} -Dcom.sun.management.jmxremote.password.file=${INSTALL_DIR}/conf/jmxremote.password"
+    CATALINA_OPTS="${CATALINA_OPTS} -Dcom.sun.management.jmxremote.access.file=${INSTALL_DIR}/conf/jmxremote.access"
+  else
+    CATALINA_OPTS="${CATALINA_OPTS} -Dcom.sun.management.jmxremote.authenticate=false"
+  fi
+fi
+
+# Access log configuration
+if [ "${ACCESS_LOG_ENABLED}" = "true" ]; then
+  # Add a new valve (just before the end of Host)
+  xmlstarlet ed -L -s "/Server/Service/Engine/Host" -t elem -n "ValveTMP" -v "" \
+    -i "//ValveTMP" -t attr -n "className" -v "org.apache.catalina.valves.AccessLogValve" \
+    -i "//ValveTMP" -t attr -n "pattern" -v "combined" \
+    -i "//ValveTMP" -t attr -n "directory" -v "logs" \
+    -i "//ValveTMP" -t attr -n "prefix" -v "access" \
+    -i "//ValveTMP" -t attr -n "suffix" -v ".log" \
+    -i "//ValveTMP" -t attr -n "rotatable" -v "true" \
+    -i "//ValveTMP" -t attr -n "renameOnRotate" -v "true" \
+    -i "//ValveTMP" -t attr -n "fileDateFormat" -v ".yyyy-MM-dd" \
+    -r "//ValveTMP" -v Valve \
+    ${INSTALL_DIR}/conf/server.xml || {
+    echo "ERROR during xmlstarlet processing (adding AccessLogValve)"
+    exit 1
+  }
+fi
+
+# -----------------------------------------------------------------------------
+# LOG GC configuration
+# -----------------------------------------------------------------------------
+if [ "${JVM_LOG_GC_ENABLED}" = "true" ]; then
+  LOG_DIR=${INSTALL_DIR}/logs
+  # -XX:+PrintGCDateStamps : print the absolute timestamp in the log statement (i.e. “2014-11-18T16:39:25.303-0800”)
+  # -XX:+PrintGCTimeStamps : print the time when the GC event started, relative to the JVM startup time (unit: seconds)
+  # -XX:+PrintGCDetails    : print the details of how much memory is reclaimed in each generation
+  JVM_LOG_GC_OPTS="-XX:+PrintGC -XX:+PrintGCDetails -XX:+PrintGCTimeStamps -XX:+PrintGCDateStamps"
+  echo "Enabling JVM GC logs with [${JVM_LOG_GC_OPTS}] options ..."
+  CATALINA_OPTS="${CATALINA_OPTS} ${JVM_LOG_GC_OPTS} -Xloggc:${LOG_DIR}/gc.log"
+  # log rotation to backup previous log file (we don't use GC Log file rotation options because they are not suitable)
+  # create the directory for older GC log file
+  [ ! -d ${LOG_DIR}/gc/ ] && mkdir ${LOG_DIR}/gc/
+  if [ -f ${LOG_DIR}/gc.log ]; then
+    JVM_LOG_GC_ARCHIVE="${LOG_DIR}/gc/gc_$(date -u +%F_%H%M%S%z).log"
+    mv ${LOG_DIR}/gc.log ${JVM_LOG_GC_ARCHIVE}
+    echo "previous JVM GC log file archived to ${JVM_LOG_GC_ARCHIVE}."
+  fi
+  echo "JVM GC logs configured and available at ${LOG_DIR}/platform-gc.log"
+fi
+
+# -----------------------------------------------------------------------------
+# JVM configuration
+# -----------------------------------------------------------------------------
+export CATALINA_OPTS="${CATALINA_OPTS} -Xmx${JVM_SIZE_MAX} -Xms${JVM_SIZE_MIN}  -XX:MaxMetaspaceSize=${JVM_METASPACE_SIZE_MAX}"
```

Comment:
The first `CATALINA_OPTS` variable usage could raise an error due to the script settings (`#!/bin/bash -eu`)

```suggestion
export CATALINA_OPTS="${CATALINA_OPTS:-} -Xmx${JVM_SIZE_MAX} -Xms${JVM_SIZE_MIN}  -XX:MaxMetaspaceSize=${JVM_METASPACE_SIZE_MAX}"
```